src/scripts/main.js

```diff
@@ -1,5 +1,35 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  const spider = document.querySelector('.spider');
+  const wall = document.querySelector('.wall');
+  const wallRect = wall.getBoundingClientRect();
+
+  if (!e.target.closest('.wall')) {
+    return;
+  }
+
+  let coordX =
```

Comment:
The variable `coordX` is being calculated using several properties. It's a good practice to add a comment explaining the calculation logic for future reference and better code readability.